### PR TITLE
feat: postgres datastore implementation - part 2

### PR DESCRIPTION
**Pull request by mfiedorowicz**
- consolidate ingestion logs and change sets repositories into single one (as related to each other)
- remove redis migrations
- remove writing ingestion logs to redis
- refactor postgres schemas
- implement postgres repository

Screenshot 2024-12-20 at 16 43 38

Integration tests:
Screenshot 2024-12-20 at 16 32 29

### CODE REVIEW

File: diode-server/reconciler/ingestion_processor.go [hunk 1]
```diff
@@ -416,13 +386,13 @@ func (p *IngestionProcessor) CreateIngestionLogs(ctx context.Context, ingestReq
 			continue
 		}
 
-		ingestionLogID := ksuid.New().String()
+		ingestionLogKSUID := ksuid.New().String()
```
Comment (leoparente) [inline diode-server/reconciler/ingestion_processor.go]: [Question] Why we are using ksuid here and then we change in line 407 to uuid?

**Comment by leoparente** [comment]
[Question] did you run go mod tidy after changes?

File: diode-server/reconciler/ingestion_processor.go (hunk 1 above)
Comment (mfiedorowicz) [reply diode-server/reconciler/ingestion_processor.go]: Good spot, forgot to make clean up on redis related things.

File: diode-server/dbstore/postgres/repository.go [hunk 2]
```diff
@@ -0,0 +1,255 @@
+package postgres
```
Comment (ltucker) [inline diode-server/dbstore/postgres/repository.go:1]: the insides of this don't have much in the way of coverage, we could consider adding something like [embedded postgres](https://github.com/fergusstrange/embedded-postgres) to get more interesting tests of the system I think. It's not perfect / has quirks, but it is realistic :D 

may not be worthwhile immediately especially if you're considering changing the interface.

Comment (mfiedorowicz) [reply diode-server/dbstore/postgres/repository.go:1]: Agree, coverage dropped overall by ~5%, to be addressed separately 👍

**Comment by ltucker** [comment]
I don't see any showstoppers here, I've been using / looking over this branch for the last few days without issue.
Especially in the interest of not making this any larger, I'm fine with this on the feature branch and addressing additional things with followups.

**Review by ltucker — APPROVED** [review]

File: diode-server/reconciler/repository.go [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+)
+
+// Repository is an interface for interacting with ingestion logs and change sets.
+type Repository interface {
+	CreateIngestionLog(ctx context.Context, ingestionLog *reconcilerpb.IngestionLog, sourceMetadata []byte) (*int32, error)
+	UpdateIngestionLogStateWithError(ctx context.Context, id int32, state reconcilerpb.State, ingestionError *reconcilerpb.IngestionError) error
+	RetrieveIngestionLogs(ctx context.Context, filter *reconcilerpb.RetrieveIngestionLogsRequest, limit int32, offset int32) ([]*reconcilerpb.IngestionLog, error)
+	CountIngestionLogsPerState(ctx context.Context) (map[reconcilerpb.State]int32, error)
+	CreateChangeSet(ctx context.Context, changeSet changeset.ChangeSet, ingestionLogID int32) (*int32, error)
+}
```
Comment (ltucker) [inline diode-server/reconciler/repository.go:17]: discussed a bit offline, would like to see this continue to evolve, possibly bury dbids inside here and/or think about the structures that go in and out separately from the grpc interface and how to address transactionality.

**Pull request merged**